src/Features/VisualBasic/Portable/ConvertIfToSwitch/VisualBasicConvertIfToSwitchCodeRefactoringProvider.Rewriting.vb

```diff
@@ -29,7 +29,14 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.ConvertIfToSwitch
 
         Public Overrides Function AsSwitchSectionStatements(operation As IOperation) As IEnumerable(Of SyntaxNode)
             Dim node = operation.Syntax
-            Return If(node.IsStatementContainerNode(), node.GetStatements(), SpecializedCollections.SingletonEnumerable(node))
+            Dim statements = If(node.IsStatementContainerNode(), DirectCast(node.GetStatements(), IEnumerable(Of SyntaxNode)).ToArray(), {node})
+
+            Dim lastStatement = statements.LastOrDefault()
+            If lastStatement IsNot Nothing Then
+                statements(statements.Length - 1) = lastStatement.WithAppendedTrailingTrivia(lastStatement.GetLastToken().GetNextToken().LeadingTrivia)
+            End If
```

Comment:
in the vb case, wouldn't this duplicate since you'recopying the trivia to something else, but not removing from teh thing you're copying from?

Reply:
The thing it's copying from is the `ElseIf` or `End If` statement which is currently being deleted.